LayoutTests/fast/canvas/toDataURL-alpha-permutation.html

```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE html> 
```

Comment:
Does this test fail without this patch? Does this patch fix a correctness issue or it is memory and perf optimization?

Reply:
This test was added due to the removal of kCGImageAlphaNoneSkipLast. This patch is primarily a memory optimization, but we want to make sure it still works for different alpha values.